### PR TITLE
Add pre-commit to repo.

### PR DESCRIPTION
**Pull request by manugoyal**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    rev: v2.2.5
+    hooks:
+      - id: codespell
+        exclude: >
+          (?x)^(
+              .*\.(json|prisma|svg)
+          )$
+        args: ["-L rouge,coo,couldn,unsecure"]
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.1
```
Comment (ankrgyl) [inline .pre-commit-config.yaml:17]: since this is new, could we use the newer version of prettier (3.1.0)?

File: apis/vercel/pnpm-lock.yaml
Comment (ankrgyl) [inline apis/vercel/pnpm-lock.yaml:1]: We should exclude this file from pre-commit hooks (it gets expensive to reformat it on every commit)

File: pnpm-lock.yaml
Comment (ankrgyl) [inline pnpm-lock.yaml:1]: we should exclude this file

File: apis/vercel/pnpm-lock.yaml
Comment (ankrgyl) [reply apis/vercel/pnpm-lock.yaml:1]: also i _think_ we might be able to delete this file

Comment (manugoyal) [reply apis/vercel/pnpm-lock.yaml:1]: I removed it from the hooks for now. Will try deleting it in another PR.

Comment (manugoyal) [reply apis/vercel/pnpm-lock.yaml:1]: nvm just did it in this PR. Seems to work locally

**Pull request merged**